frontend/src/pages/Publish.tsx

```diff
@@ -13,6 +13,22 @@ import GenerateAIBtn from '../components/GenerateAIBtn';
 import PublishTags from '../components/PublishTags';
 import { htmlTagRegex } from '../util/string';
 
+
+// Register custom video handler with Quill
+const videoHandler = function (this: Quill) {
+  const range = this.getSelection();
+  const value = prompt('Please enter YouTube URL:');
+  if (value) {
+    this.insertEmbed(range.index, 'video', value, 'user');
+    this.setSelection(range.index + 1);
+  }
+};
```

Comment:
We can make videoHandler reusuable. 
Define it once in a util/lib file and use it in both files.